index.html

```diff
@@ -296,141 +292,176 @@
+      case "H": // H period
+        for (var k = 0; k < schedule["classes"].length; k++) {
+          if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
+            if (schedule["classes"][k]["name"] != "Free Period") {
+              var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
+              var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
+              scheduleObj = { name: schedule["classes"][k]["name"],
+                teacher: teacherName,
+                teacherLastName: teacherLast,
+                room: schedule["classes"][k]["room"],
+                period: days[0][day][school][currentSlot]["period"],
+                time: ""};
 
-            default:
-              scheduleObj = {
-                name: days[0][day][school][currentSlot]["period"],
+            } else {
+              scheduleObj = {name: "Free Period",
                 teacher: "",
+                teacherLastName: "",
                 room: "",
-                period: "X",
-                time: ""
-              };
-              if (scheduleObj.name == "Lunch" ||
-                  scheduleObj.name == "Assembly" ||
-                  scheduleObj.name == "US Community") {
-                scheduleObj.room = "LPC Commons";
-              }
-        }
-        if (scheduleObj.teacher == "N/A") {
-          scheduleObj.teacher = "";
+                period: days[0][day][school][currentSlot]["period"],
+                time: ""};
+            }
+            break;
+          }
         }
-        if (scheduleObj.room == "N/A") {
-          scheduleObj.room = "";
+        break;
+
+      default:
+        scheduleObj = {
+          name: days[0][day][school][currentSlot]["period"],
+          teacher: "",
+          room: "",
+          period: "X",
+          time: ""
+        };
+        if (scheduleObj.name == "Lunch" ||
+          scheduleObj.name == "Assembly" ||
+          scheduleObj.name == "US Community") {
+
+          scheduleObj.room = "LPC Commons";
         }
-        // Add in photos if the user's schedule is being rendered
-        if (type == "full") {
+    }
+    if (scheduleObj.teacher == "N/A") {
+      scheduleObj.teacher = "";
+    }
+    if (scheduleObj.room == "N/A") {
+      scheduleObj.room = "";
+    }
+    return scheduleObj;
+  }
 
-          // If the user is a normal user and the class is a normal class
-          if (scheduleObj.teacher != "" && schedule.grade) {
-            var sanitized_teacher_name;
-            sanitized_teacher_name = scheduleObj.teacher.toLowerCase();
-            sanitized_teacher_name = sanitized_teacher_name.replace(/ /g,"_");
-            scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
-            scheduleObj.teacherLink = "/teacher/" + sanitized_teacher_name;
+  // Add the images on the left side to full schedules
+  function addScheduleImages(scheduleObj, userSchedule) {
```

Comment:
Probably want to call this from createClassEntry, so that function can return a fully-formed object.